Tests/Client/RequestRecorderURLProtocol.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  MockURLProtocol.swift
+//  StreamChatClientTests
+//
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import XCTest
+
+/// This URLProtocol subclass allows to intercept the network communication
+/// and provides the latest network request made.
+class RequestRecorderURLProtocol: URLProtocol {
+
+    /// Returns the latest network request this URLProtocol recorded.
+    ///
+    /// If no request has been made since the last time this function was invoked, it synchronously
+    /// waits for the next request to be made.
+    ///
+    /// - Parameter timeout: Specifies the time the function waits for a new request
+    /// to be made.
+    static func getLatestRequestSynchronously(timeout: TimeInterval) -> URLRequest? {
```

Comment:
We usually don't use `get` and `set` words. It's not swifty :)

Reply:
fixed